src/v/cluster/types.cc

```diff
@@ -1046,6 +1052,8 @@ adl<cluster::topic_configuration>::from(iobuf_parser& in) {
     // Legacy topics from pre-22.3 get remote delete disabled.
     cfg.properties.remote_delete = storage::ntp_config::legacy_remote_delete;
 
+    // TODO how to handle segment_ms here?
```

Comment:
is there a need for special treatment here?

Reply:
As long as the default initialization of the segment_ms field is what we want for legacy topics, then no.